File: 3_Pregunta/TerceraPregunta.scala
def separar_palabras: String => List[String] = (cadena: String) => cadena.split(" ").toList
def separar_letras: String => List[Char] = (cadena: String) => cadena.toCharArray.toList

println(separar_palabras("Esto es una cadena con texto Hola Mundo 1"))
println(separar_letras("Esto es una cadena con texto Hola Mundo 2"))

File: 3_Pregunta/TerceraPregunta.hs
separar_palabras :: String -> [String]
separar_palabras cadena = (\x -> words x) cadena

separar_letras :: String -> [[Char]]
separar_letras cadena = (\x -> map (:[]) x) cadena

main = do
    let stri = separar_palabras "Esto es una cadena con texto Hola Mundo 1"
    let characters = separar_letras "Esto es una cadena con texto Hola Mundo 2"
    print stri
    print characters